src/_guides/environment-declarations.md

```diff
@@ -0,0 +1,187 @@
+---
+title: Configuring apps with environment declarations
+description: >
+  Learn about using environment declarations to customize application behavior.
+---
+
+Dart supports specifying environment declarations when
+building or running an application.
+Environment declarations are used to specify
+configuration values or options as key-value pairs
+that are accessed and evaluated at compile time.
```

Comment:
The "at compile time" is why calling this a "compilation environment" should make sense to the user.